lib/elixir/lib/io.ex

```diff
@@ -214,45 +220,53 @@ defmodule IO do
   end
 
   @doc """
-  Gets a number of bytes from the io device. If the
-  io device is a unicode device, `count` implies
-  the number of unicode codepoints to be retrieved.
+  Gets a number of bytes from IO device `:stdio`.
+
+  If `:stdio` is a Unicode device, `count` implies
+  the number of Unicode codepoints to be retrieved.
   Otherwise, `count` is the number of raw bytes to be retrieved.
-  It returns:
 
-    * `data` - the input characters
+  See `IO.getn/3` for a description of return values.
 
-    * `:eof` - end of file was encountered
-
-    * `{:error, reason}` - other (rare) error condition;
-      for instance, `{:error, :estale}` if reading from an
-      NFS volume
   """
   @spec getn(chardata | String.Chars.t, pos_integer) :: chardata | nodata
   @spec getn(device, chardata | String.Chars.t) :: chardata | nodata
   def getn(prompt, count \\ 1)
 
-  def getn(prompt, count) when is_integer(count) do
+  def getn(prompt, count) when is_integer(count) and count > 0 do
     getn(group_leader, prompt, count)
   end
 
+  # This a catch-all function to call IO.getn/3
   def getn(device, prompt) do
```

Comment:
Is this clause valid for integers less than 0? If not the added guard above won't really help unless you also add `when not is_integer(prompt)` here.